<?xml version="1.0"?>

<launch>
    <arg name="device" default="/dev/input/js0"/>

    <node name="joy" pkg="ccr" type="joy_node" respawn="false" output="screen">
      <param name="joy_topic"   value="joy"/>
      <param name="vel_topic"   value="cmd_vel"/>
      <param name="mode_topic_name" value="mobile_base/command/mode"/>
      <param name="mode_cmd"    value="normal"/>
      <param name="L1_BUTTON"   value="4"/>
      <param name="R1_BUTTON"   value="5"/>
      <param name="START_BUTTON" value="9"/>
      <param name="LEFT_Y_AXIS"  value="0"/>
      <param name="LEFT_X_AXIS"  value="1"/>
      <param name="RIGHT_Y_AXIS" value="2"/>
      <param name="RIGHT_X_AXIS" value="3"/>
      <param name="X_BUTTON"     value="4"/>
      <param name="Y_BUTTON"     value="5"/>
    </node>

    <node name="joy_driver" pkg="joy" type="joy_node" respawn="false" output="screen">
	<param name="dev" value="$(arg device)"/>
	<param name="autorepeat_rate" value="4.0"/>
    </node>
</launch>
